tests/runtime/usdt

```diff
@@ -0,0 +1,92 @@
+NAME "usdt probes - list probes by file"
+RUN bpftrace -l 'usdt:./testprogs/usdt_test'
+EXPECT usdt:./testprogs/usdt_test:tracetest:testprobe
+TIMEOUT 5
+
+NAME "usdt probes - list probes by pid"
+RUN bpftrace -l 'usdt:*' -p $(pidof usdt_test)
+EXPECT usdt_test:tracetest:testprobe
+TIMEOUT 5
+BEFORE (./testprogs/usdt_test &);
+
+NAME "usdt probes - filter probes by file on provider"
+RUN bpftrace -l 'usdt:./testprogs/usdt_test:tracetest2:*'
+EXPECT usdt:./testprogs/usdt_test:tracetest2:testprobe2
+TIMEOUT 5
+
+NAME "usdt probes - filter probes by pid on provider"
+RUN bpftrace -l 'usdt:*:tracetest2:*' -p $(pidof usdt_test)
+EXPECT usdt_test:tracetest2:testprobe2
+TIMEOUT 5
+BEFORE (./testprogs/usdt_test &);
+
+NAME "usdt probes - filter probes by file and wildcard probe name"
+RUN bpftrace -l 'usdt:./testprogs/usdt_test:tracetest:test*'
+EXPECT usdt:./testprogs/usdt_test:tracetest:testprobe2
+TIMEOUT 5
+
+NAME "usdt probes - filter probes by pid and wildcard probe name"
+RUN bpftrace -l 'usdt:*:tracetest:test*' -p $(pidof usdt_test)
+EXPECT usdt_test:tracetest:testprobe2
+TIMEOUT 5
+BEFORE (./testprogs/usdt_test &);
+
+NAME "usdt probes - attach to fully specified probe by file"
+RUN bpftrace -e 'usdt:./testprogs/usdt_test:tracetest:testprobe { printf("here\n" ); exit(); }'
+EXPECT here
+TIMEOUT 5
+BEFORE (./testprogs/usdt_test &);
+
+NAME "usdt probes - attach to fully specified probe by pid"
+RUN bpftrace -e 'usdt::tracetest:testprobe { printf("here\n" ); exit(); }' -p $(pidof usdt_test)
+EXPECT here
+TIMEOUT 5
+BEFORE (./testprogs/usdt_test &);
+
+NAME "usdt probes - all probes by wildcard and file"
+RUN bpftrace -e 'usdt:./testprogs/usdt_test:* { printf("here\n" ); exit(); }'
+EXPECT Attaching 3 probes...
+TIMEOUT 5
+BEFORE (./testprogs/usdt_test &);
+
+NAME "usdt probes - all probes by wildcard and pid"
+RUN bpftrace -e 'usdt:* { printf("here\n" ); exit(); }' -p $(pidof usdt_test)
```

Comment:
I'm not sure I like allowing wildcards on `path`. Maybe the syntax here should be:

```suggestion
RUN bpftrace -e 'usdt::* { printf("here\n" ); exit(); }' -p $(pidof usdt_test)
```

Reply:
This is similar to something @tyroguru suggested, but would require a larger overhaul of how we parse wildcards in listing probes. What you indicate is closer to dtrace, and what most people would expect. To geth the same behavior as a `::` in bpftrace, I believe you need to explicitly put in `:*`.

This is confusing, and even inconsistent within bpftrace itself because the way that we accept probe definitions takes the colon / dtrace form, not needing the asterix to wildcard/glob. Listing and attaching should clearly use the same notation here, ideally using the same underlying calls to filter - ensuring consistency. 

We should cut another issue for this.